ArduCopter/control_auto.cpp

```diff
@@ -361,8 +361,10 @@ void Copter::auto_land_start(const Vector3f& destination)
     wp_nav.init_loiter_target(destination);
 
     // initialise position and desired velocity
-    pos_control.set_alt_target(inertial_nav.get_altitude());
-    pos_control.set_desired_velocity_z(inertial_nav.get_velocity_z());
+    if (!pos_control.is_active_z()) {
+        pos_control.set_alt_target(inertial_nav.get_altitude());
```

Comment:
I think the difference is that in auto mode (or most sub modes of auto) the AC_WPNav library is responsible for calling the vertical position controller.  In most of the other flight modes, (even Loiter) the flight mode code (like control_loiter.cpp) is responsible for calling the vertical position controller.  This is why the initialisation is a bit different except for Auto's land mode.